### PR TITLE
[PLAT-695] Add trending notifications to valid_types

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Adds trending-playlist and trending-underground notification types to discovery-notifications "valid_types" parameter
- Refactors `fetchNotification` function to more accurately switch between new and old notification format.

### CODE REVIEW

File: packages/web/src/common/store/notifications/fetchNotifications.ts [hunk 1]
```diff
@@ -20,106 +19,132 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
...
-    FeatureFlags.DISCOVERY_NOTIFICATIONS
+    FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
+  )
+  const isSaveOfRepostEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
+  )
+  const isTrendingPlaylistEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
```
Comment (sabrina-kiam) [inline packages/web/src/common/store/notifications/fetchNotifications.ts:82]: is this checking the wrong feature flag??

File: packages/web/src/common/store/notifications/fetchNotifications.ts [hunk 2]
```diff
@@ -20,106 +19,132 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
...
   )
+  const isTrendingUndergroundEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
```
Comment (sabrina-kiam) [inline packages/web/src/common/store/notifications/fetchNotifications.ts:86]: same here i think this is wrong feature flag?

File: packages/web/src/common/store/notifications/fetchNotifications.ts [hunk 3]
```diff
@@ -20,106 +19,132 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
...
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const remoteConfig = yield* getContext('remoteConfigInstance')
+
+  const isRepostOfRepostEnabled = yield* call(
     getFeatureEnabled,
-    FeatureFlags.DISCOVERY_NOTIFICATIONS
+    FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
+  )
+  const isSaveOfRepostEnabled = yield* call(
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/common/store/notifications/fetchNotifications.ts:76]: oh yess i totally hear you, i like this!

File: packages/web/src/common/store/notifications/fetchNotifications.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/common/store/notifications/fetchNotifications.ts:82]: yes RIPP

File: packages/web/src/common/store/notifications/fetchNotifications.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/common/store/notifications/fetchNotifications.ts:76]: i just tried and it's annoyingly just as long :/ ideally wed have a "getFeatures" function and then that returns an object than we can access. even though it's a lot of redundant code at least it's easy to understand? but yeah i hear you not the best...

[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/common/store/notifications/fetchNotifications.ts:76]: new method: 
```
  const validTypesFlags = [
    FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS,
    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS,
    FeatureFlags.TRENDING_PLAYLIST_NOTIFICATIONS,
    FeatureFlags.TRENDING_UNDERGROUND_NOTIFICATIONS
  ]

  const validTypesMap: { [flag in FeatureFlags]?: boolean } = {}

  for (const validTypeFlag of validTypesFlags) {
    validTypesMap[validTypeFlag] = yield* call(getFeatureEnabled, validTypeFlag)
  }

  const validTypes = [
    validTypesMap[FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS]
      ? 'repost_of_repost'
      : null,

    validTypesMap[FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS]
      ? 'save_of_repost'
      : null,

    validTypesMap[FeatureFlags.TRENDING_PLAYLIST_NOTIFICATIONS]
      ? 'trending_playlist'
      : null,

    validTypesMap[FeatureFlags.TRENDING_UNDERGROUND_NOTIFICATIONS]
      ? 'trending_underground'
      : null
  ].filter(removeNullable)
```

old method: 
```
  const isRepostOfRepostEnabled = yield* call(
    getFeatureEnabled,
    FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
  )
  const isSaveOfRepostEnabled = yield* call(
    getFeatureEnabled,
    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
  )
  const isTrendingPlaylistEnabled = yield* call(
    getFeatureEnabled,
    FeatureFlags.TRENDING_PLAYLIST_NOTIFICATIONS
  )
  const isTrendingUndergroundEnabled = yield* call(
    getFeatureEnabled,
    FeatureFlags.TRENDING_UNDERGROUND_NOTIFICATIONS
  )

  const validTypes = [
    isRepostOfRepostEnabled ? 'repost_of_repost' : null,
    isSaveOfRepostEnabled ? 'save_of_repost' : null,
    isTrendingPlaylistEnabled ? 'trending_playlist' : null,
    isTrendingUndergroundEnabled ? 'trending_underground' : null
  ].filter(removeNullable)
```

[the remark this answers is not in the document]
Comment (sabrina-kiam) [reply packages/web/src/common/store/notifications/fetchNotifications.ts:76]: damn jk then lol you can leave however you prefer in that case!

**Review by sabrina-kiam — APPROVED** [review]

**Pull request merged**